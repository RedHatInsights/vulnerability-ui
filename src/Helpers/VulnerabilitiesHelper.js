import { parseCvssScore } from '@red-hat-insights/insights-frontend-components/Utilities/helpers';
import * as MiscHelper from './MiscHelper';
import React from 'react';
import ImpactBadge from '../Components/PresentationalComponents/ImpactBadge/ImpactBadge';

export function createCveListByAccount(cveList) {
    let isLoading = cveList.isLoading;
    if (!isLoading) {
        let payload = cveList.payload.asMutable();
        let data = payload.data;
        let meta = payload.meta;

        data = data.map(row => ({
            id: row.id,
            cells: [
                <ImpactBadge impact={row.attributes.impact} hastooltip={true} key={row.id.toString()} />,
                row.attributes.synopsis,
                MiscHelper.truncate(row.attributes.description, 200),
                parseCvssScore(row.attributes.cvss2_score, row.attributes.cvss3_score),
                row.attributes.systems_affected,
                MiscHelper.processDate(row.attributes.public_date)
            ]
        }));
        return { data, meta, isLoading };
    }

    return { data: [], meta: (cveList.payload && cveList.payload.meta) || {}, isLoading };
}

export function createCurrentVulnerabilitiesCard(impacts) {
    let dataSet = impacts.payload;
    let isLoading = impacts.isLoading;
    if (!isLoading) {
        dataSet = impacts.payload;
    }

    dataSet = { content: dataSet, isLoading };
    return dataSet;
}
