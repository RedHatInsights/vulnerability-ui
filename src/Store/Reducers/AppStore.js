import * as ActionTypes from '../ActionTypes';

// Initial State
export const initialState = {
    popup: {}
};

// Reducer
export const AppStore = (state = initialState, action) => {
    switch (action.type) {
        case ActionTypes.TOGGLE_POPUP: {
            const { page, state } = action.payload;
            return {
                ...state,
                popup: {
                    ...state.popup,
                    [page]: state
                }
            };
        }

        default:
            return state;
    }
};
