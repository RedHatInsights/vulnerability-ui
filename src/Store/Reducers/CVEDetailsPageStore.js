import Immutable from 'seamless-immutable';
import * as ActionTypes from '../ActionTypes';

// Initial State
export const initialState = Immutable({
    affectedSystemsByCVE: {
        isLoading: true,
        openedRows: [],
        payload: {
            meta: {
                total_items: 0,
                page: 1,
                page_size: 20
            }
        }
    },
    parameters: {
        page: 1,
        page_size: 20,
        selectedHosts: [],
        status_id: undefined,
        sort: '-last_upload',
        show_advisories: true
    },
    cveDetails: {
        isLoading: true,
        payload: {},
        error: false
    }
});

// Reducer
export const CVEDetailsPageStore = (state = initialState, action) => {
    let newState;
    switch (action.type) {
        case ActionTypes.FETCH_CVE_DETAILS + '_REJECTED':
            newState = state.setIn(['cveDetails', 'error'], true);
            newState = newState.setIn(['cveDetails', 'isLoading'], false);
            return newState;
        case ActionTypes.FETCH_CVE_DETAILS + '_PENDING':
            newState = state.setIn(['cveDetails', 'isLoading'], true);
            return newState;

        case ActionTypes.FETCH_CVE_DETAILS + '_FULFILLED':
            newState = state.setIn(['cveDetails', 'payload'], action.payload);
            newState = newState.setIn(['cveDetails', 'isLoading'], false);
            return newState;

        case ActionTypes.FETCH_AFFECTED_SYSTEMS_BY_CVE + '_PENDING':
            newState = state.setIn(['affectedSystemsByCVE', 'isLoading'], true);
            return newState;

        case ActionTypes.FETCH_AFFECTED_SYSTEMS_BY_CVE + '_FULFILLED':
            newState = state.setIn(['affectedSystemsByCVE', 'payload'], action.payload);
            newState = newState.setIn(['affectedSystemsByCVE', 'isLoading'], false);
            return newState;
        case ActionTypes.CHANGE_EXPOSED_SYSTEMS_PARAMETERS:
            newState = state.setIn(['parameters'], {
                ...state.parameters,
                ...action.payload,
                page_size: action.payload.page_size || state.parameters.page_size
            });
            return newState;
        case ActionTypes.SET_GLOBAL_FILTER:
            newState = state.setIn(['parameters'], {
                ...state.parameters,
                ...action.payload
            });
            return newState;
        case ActionTypes.CLEAR_CVE_STORE:
            return initialState;
        default:
            return state;
    }
};

