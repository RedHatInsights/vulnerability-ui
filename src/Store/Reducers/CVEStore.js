import Immutable from 'seamless-immutable';
import * as ActionTypes from '../../Store/ActionTypes';

// Initial State
export const initialState = Immutable({
    affectedSystemsByCVE: {
        isLoading: true,
        payload: [],
        openedRows: []
    },
    cveDetails: {
        isLoading: true,
        payload: {},
        error: false
    }
});

// Reducer
export const CVEReducer = (state = initialState, action) => {
    let newState;
    let openedRows;
    switch (action.type) {
        case ActionTypes.HANDLE_CVE_DETAILS_ERROR:
            newState = state.setIn(['cveDetails', 'error'], action.payload.error);
            newState = newState.setIn(['cveDetails', 'isLoading'], false);
            return newState;
        case ActionTypes.FETCH_CVE_DETAILS + '_PENDING':
            newState = state.setIn(['cveDetails', 'isLoading'], true);
            return newState;

        case ActionTypes.FETCH_CVE_DETAILS + '_FULFILLED':
            newState = state.setIn(['cveDetails', 'payload'], action.payload);
            newState = newState.setIn(['cveDetails', 'isLoading'], false);
            return newState;

        case ActionTypes.FETCH_AFFECTED_SYSTEMS_BY_CVE + '_PENDING':
            newState = state.setIn(['affectedSystemsByCVE', 'isLoading'], true);
            return newState;

        case ActionTypes.FETCH_AFFECTED_SYSTEMS_BY_CVE + '_FULFILLED':
            newState = state.setIn(['affectedSystemsByCVE', 'payload'], action.payload);
            newState = newState.setIn(['affectedSystemsByCVE', 'isLoading'], false);
            return newState;
        case ActionTypes.EXPAND_ROW:
            openedRows = [...state.affectedSystemsByCVE.openedRows];

            if (openedRows.includes(action.payload.id)) {
                openedRows.splice(openedRows.indexOf(action.payload.id), 1);
            } else {
                openedRows = [action.payload.id, ...openedRows];
            }

            newState = state.setIn(['affectedSystemsByCVE', 'openedRows'], openedRows);
            return newState;
        default:
            return state;
    }
};
