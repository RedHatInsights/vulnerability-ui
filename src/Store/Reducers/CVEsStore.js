import * as ActionTypes from '../ActionTypes';
import { applyGlobalFilter, isTimestampValid } from './reducersHelper';

// Initial State
export const initialState = {
    cveList: {
        isLoading: true,
        payload: [],
        timestamp: new Date()
    },
    parameters: {
        sort: '-public_date',
        affecting: 'true,false'
    },
    selectedCves: [],
    expandedRows: [],
    isAllExpanded: false
};

// Reducer
export const CVEsStore = (state = initialState, action) => {
    let newState = { ...state };

    switch (action.type) {
        case ActionTypes.FETCH_CVE_LIST + '_PENDING':
            newState.cveList = {
                timestamp: action.meta,
                isLoading: true
            };
            return newState;

        case ActionTypes.FETCH_CVE_LIST + '_REJECTED':
            newState.error = true;
            newState.cveList = {
                isLoading: false,
                payload: { errors: action.payload }
            };
            return newState;

        case ActionTypes.FETCH_CVE_LIST + '_FULFILLED':
            if (isTimestampValid(state.cveList.timestamp, action.meta)) {
                newState = {
                    ...state,
                    cveList: {
                        ...state.cveList,
                        payload: action.payload,
                        isLoading: false
                    },
                    ...state.isAllExpanded && { expandedRows: action.payload.data.map(({ id }) => id) }
                };

                return newState;
            }

            return state;

        case ActionTypes.CHANGE_CVE_LIST_PARAMETERS + '_FULFILLED':
            newState.parameters = { ...newState.parameters, ...action.payload };
            return newState;

        case ActionTypes.SELECT_CVE: {
            let selectedCves = newState.selectedCves.slice();
            if (Array.isArray(action.payload)) {
                selectedCves = action.payload;
            } else {
                (selectedCves.includes(action.payload) && selectedCves.splice(selectedCves.indexOf(action.payload), 1)) ||
                    selectedCves.push(action.payload);
            }

            return { ...newState, selectedCves  };
        }

        case ActionTypes.EXPAND_CVE: {
            if (Array.isArray(action.payload)) {
                const expandedRows = action.payload;
                const isAllExpanded = action.payload.length === 0 ? false : true;
                return { ...newState,  expandedRows, isAllExpanded };
            }

            const cveName = newState.cveList.payload.data[action.payload / 2].id;
            const expandedRows = newState.expandedRows.slice();
            expandedRows.includes(cveName) && expandedRows.splice(expandedRows.indexOf(cveName), 1)
            || expandedRows.push(cveName);

            return { ...newState, expandedRows };
        }

        case ActionTypes.CLEAR_CVES_STORE:
            return initialState;

        case ActionTypes.SET_GLOBAL_FILTER:
            newState.parameters = {
                ...applyGlobalFilter(newState.parameters, action.payload)
            };
            return newState;

        default:
            return state;
    }
};
