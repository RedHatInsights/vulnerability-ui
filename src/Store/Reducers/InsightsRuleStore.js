import * as ActionTypes from '../ActionTypes';

const initialState = {
    error: false,
    isLoading: true,
    systemRules: []
};

export const InsightsRuleStore = (state = initialState, action) => {
    let newState = { ...state };
    switch (action.type) {
        case ActionTypes.FETCH_INSIGHTS_SYSTEM_RULES + '_PENDING':
            return newState;
        case ActionTypes.FETCH_INSIGHTS_SYSTEM_RULES + '_REJECTED':
            newState.isLoading = false;
            newState.error = true;
            return newState;
        case ActionTypes.FETCH_INSIGHTS_SYSTEM_RULES + '_FULFILLED':
            newState.isLoading = false;
            newState.systemRules = action.payload;
            return newState;
        default:
            return state;
    }
};
