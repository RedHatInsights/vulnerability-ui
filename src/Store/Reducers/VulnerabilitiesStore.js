import * as ActionTypes from '../ActionTypes';

// Initial State
export const initialState = {
    cveList: {
        isLoading: true,
        payload: [],
        timestamp: new Date()
    },
    parameters: {
        show_all: 'true',
        sort: '-public_date',
        selectedCves: [],
        openedCves: [],
        expandCveDescription: false
    }
};

function isValid(stateTimestamp, actionTimestamp) {
    return actionTimestamp >= stateTimestamp;
}

// Reducer
export const VulnerabilitiesStore = (state = initialState, action) => {
    let newState = { ...state };

    switch (action.type) {
        case ActionTypes.FETCH_CVE_LIST + '_PENDING':
            newState.cveList.isLoading = true;
            newState.cveList.timestamp = action.meta;
            return newState;

        case ActionTypes.FETCH_CVE_LIST + '_REJECTED':
            newState.cveList.isLoading = false;
            newState.cveList.payload = { errors: action.payload };
            return newState;

        case ActionTypes.FETCH_CVE_LIST + '_FULFILLED':
            if (isValid(state.cveList.timestamp, action.meta)) {
                newState.cveList.payload = action.payload;
                newState.cveList.isLoading = false;
                newState.cveList.timestamp = action.meta;
                return newState;
            }

            return state;

        case ActionTypes.CHANGE_CVE_LIST_PARAMETERS + '_FULFILLED':
            newState.parameters = { ...newState.parameters, ...action.payload };
            return newState;

        case ActionTypes.SELECT_CVE: {
            let selectedCves = newState.parameters.selectedCves.slice();
            if (Array.isArray(action.payload)) {
                selectedCves = action.payload;
            } else {
                (selectedCves.includes(action.payload) && selectedCves.splice(selectedCves.indexOf(action.payload), 1)) ||
                    selectedCves.push(action.payload);
            }

            return { ...newState, parameters: { ...newState.parameters, selectedCves } };
        }

        case ActionTypes.EXPAND_CVE: {
            if (Array.isArray(action.payload)) {
                const openedCves = action.payload;
                const expandCveDescription = action.payload.length === 0 ? false : true;
                return { ...newState, parameters: { ...newState.parameters, openedCves, expandCveDescription } };
            }

            const cveName = newState.cveList.payload.data[action.payload / 2].id;
            const openedCves = newState.parameters.openedCves.slice();
            (openedCves.includes(cveName) && openedCves.splice(openedCves.indexOf(cveName), 1)) || openedCves.push(cveName);
            return { ...newState, parameters: { ...newState.parameters, openedCves } };
        }

        default:
            return state;
    }
};
