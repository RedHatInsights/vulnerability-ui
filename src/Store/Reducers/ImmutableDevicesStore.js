import Immutable from 'seamless-immutable';
import * as ActionTypes from '../ActionTypes';
import { applyGlobalFilter } from './reducersHelper';
import { DEFAULT_PAGE_SIZE } from '../../Helpers/constants';

export const initialState = Immutable({
    parameters: {
        page: 1,
        page_size: DEFAULT_PAGE_SIZE,
        sort: '-updated',
        //makes sure only edge devices are shown
        host_type: 'edge'
    },
    devices: {
        isLoading: true,
        payload: {},
        error: {
            hasError: false
        }
    }
});

export const ImmutableDevicesStore = (state = initialState, action) => {
    let newState;

    switch (action.type) {
        case ActionTypes.FETCH_CVE_DEVICES + '_REJECTED':
            newState = state.setIn(['cveDetails', 'error'], { hasError: true, errorCode: action.payload.status });
            newState = newState.setIn(['cveDetails', 'isLoading'], false);
            return newState;
        case ActionTypes.FETCH_CVE_DEVICES + '_PENDING':
            newState = state.setIn(['cveDetails', 'isLoading'], true);
            return newState;

        case ActionTypes.FETCH_CVE_DEVICES + '_FULFILLED':
            newState = state.setIn(['cveDetails', 'payload'], action.payload);
            newState = newState.setIn(['cveDetails', 'isLoading'], false);
            return newState;

        case ActionTypes.CHANGE_EXPOSED_DEVICES_PARAMETERS: {
            const { reset, ...params } = action.payload;

            newState = state.setIn(['parameters'], {
                ...!reset ? state.parameters : {},
                ...params,
                page_size: action.payload.page_size || state.parameters.page_size
            });
            return newState;
        }

        case ActionTypes.SET_GLOBAL_FILTER:
            newState = state.setIn(['parameters'], {
                ...applyGlobalFilter(state.parameters, action.payload)
            });
            return newState;
        case ActionTypes.CLEAR_DEVICES_STORE:
            return initialState;
        default:
            return state;
    }
};

