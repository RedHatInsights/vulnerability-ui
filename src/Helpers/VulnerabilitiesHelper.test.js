import { initialState } from '../Store/Reducers/VulnerabilitiesStore';
import { createCveListByAccount } from './VulnerabilitiesHelper';

const { parameters } = initialState;
describe('VulnerabilitiesHelper', () => {
    it('test function createCveListByAccount() with cveList param', () => {
        const cveList = {
            isLoading: false,
            payload: {
                meta: {
                    page: 1,
                    page_size: 20,
                    total_items: 248,
                    pages: 10,
                    sort: '-public_date',
                    filter: '',
                    show_all: 'false',
                    cvss_from: '',
                    cvss_to: '',
                    public_from: '',
                    public_to: ''
                },
                data: [
                    {
                        type: 'cve',
                        id: 'CVE-2019-6454',
                        attributes: {
                            systems_affected: 2,
                            synopsis: 'CVE-2019-6454',
                            public_date: '2019-02-18T00:00:00+00:00',
                            impact: 'Important',
                            description: '** RESERVED ** This candidate',
                            cvss2_score: '',
                            cvss3_score: '7.000',
                            business_risk_id: '3',
                            status_id: 2,
                            cve_status_id: 2
                        }
                    },
                    {
                        type: 'cve',
                        id: 'CVE-2019-6116',
                        attributes: {
                            systems_affected: 2,
                            synopsis: 'CVE-2019-6116',
                            public_date: '2019-01-23T00:00:00+00:00',
                            impact: 'Important',
                            description: 'It was found that ghostscript',
                            cvss2_score: '',
                            cvss3_score: '7.300',
                            business_risk_id: '1',
                            status_id: 1,
                            cve_status_id: 2
                        }
                    },
                    {
                        type: 'cve',
                        id: 'CVE-2019-3815',
                        attributes: {
                            systems_affected: 2,
                            synopsis: 'CVE-2019-3815',
                            public_date: '2019-01-14T00:00:00+00:00',
                            impact: 'Low',
                            description: 'A memory leak was discovered',
                            cvss2_score: '',
                            cvss3_score: '3.300',
                            business_risk_id: '2',
                            status_id: 2,
                            cve_status_id: 2
                        }
                    }
                ]
            }
        };
        expect(JSON.stringify(createCveListByAccount({ cveList, parameters }))).toMatchSnapshot();
    });

    it('VulnerabilitiesHelper - loading', () => {
        const cve_list = createCveListByAccount({
            cveList: { isLoading: true, payload: { meta: 'testObject', data: 'testData' }, parameters }
        });
        expect(cve_list).toEqual({ data: [], meta: 'testObject', isLoading: true });
    });

    it('VulnerabilitiesHelper - loading, empty meta', () => {
        const cve_list = createCveListByAccount({ cveList: { isLoading: true, payload: { data: 'testData' } }, parameters });
        expect(cve_list).toEqual({ data: [], meta: {}, isLoading: true });
    });
});
