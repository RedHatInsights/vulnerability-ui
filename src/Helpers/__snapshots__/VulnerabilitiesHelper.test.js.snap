// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VulnerabilitiesHelper test function createCveListByAccount() with cveList param 1`] = `"{\\"data\\":[{\\"id\\":\\"CVE-2019-6454\\",\\"business_risk_id\\":\\"3\\",\\"status_id\\":2,\\"cells\\":[{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6454\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/cves/CVE-2019-6454\\",\\"children\\":\\"CVE-2019-6454\\",\\"replace\\":false},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6454\\",\\"ref\\":null,\\"props\\":{\\"children\\":\\"18 Feb 2019\\"},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6454\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"impact\\":\\"Important\\",\\"hasLabel\\":true,\\"size\\":\\"sm\\",\\"hasTooltip\\":true},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6454\\",\\"ref\\":null,\\"props\\":{\\"children\\":\\"7.0\\"},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6454\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"a\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"href\\":\\"http://localhost/rhel/vulnerability/cves/CVE-2019-6454\\",\\"children\\":\\"2\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6454\\",\\"ref\\":null,\\"props\\":{\\"children\\":\\"High\\"},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6454\\",\\"ref\\":null,\\"props\\":{\\"children\\":[\\"\\",\\" \\",\\"On-hold\\"]},\\"_owner\\":null,\\"_store\\":{}}}],\\"isOpen\\":false,\\"selected\\":false},{\\"cells\\":[{\\"title\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":[{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Description\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"component\\":\\"p\\",\\"children\\":\\"** RESERVED ** This candidate\\"},\\"_owner\\":null,\\"_store\\":{}}]},\\"_owner\\":null,\\"_store\\":{}}}],\\"parent\\":0},{\\"id\\":\\"CVE-2019-6116\\",\\"business_risk_id\\":\\"1\\",\\"status_id\\":1,\\"cells\\":[{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6116\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/cves/CVE-2019-6116\\",\\"children\\":\\"CVE-2019-6116\\",\\"replace\\":false},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6116\\",\\"ref\\":null,\\"props\\":{\\"children\\":\\"23 Jan 2019\\"},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6116\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"impact\\":\\"Important\\",\\"hasLabel\\":true,\\"size\\":\\"sm\\",\\"hasTooltip\\":true},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6116\\",\\"ref\\":null,\\"props\\":{\\"children\\":\\"7.3\\"},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6116\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"a\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"href\\":\\"http://localhost/rhel/vulnerability/cves/CVE-2019-6116\\",\\"children\\":\\"2\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6116\\",\\"ref\\":null,\\"props\\":{\\"children\\":\\"Low\\"},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-6116\\",\\"ref\\":null,\\"props\\":{\\"children\\":[\\"\\",\\" \\",\\"In-review\\"]},\\"_owner\\":null,\\"_store\\":{}}}],\\"isOpen\\":false,\\"selected\\":false},{\\"cells\\":[{\\"title\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":[{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Description\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"component\\":\\"p\\",\\"children\\":\\"It was found that ghostscript\\"},\\"_owner\\":null,\\"_store\\":{}}]},\\"_owner\\":null,\\"_store\\":{}}}],\\"parent\\":2},{\\"id\\":\\"CVE-2019-3815\\",\\"business_risk_id\\":\\"2\\",\\"status_id\\":2,\\"cells\\":[{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-3815\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/cves/CVE-2019-3815\\",\\"children\\":\\"CVE-2019-3815\\",\\"replace\\":false},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-3815\\",\\"ref\\":null,\\"props\\":{\\"children\\":\\"14 Jan 2019\\"},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-3815\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"impact\\":\\"Low\\",\\"hasLabel\\":true,\\"size\\":\\"sm\\",\\"hasTooltip\\":true},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-3815\\",\\"ref\\":null,\\"props\\":{\\"children\\":\\"3.3\\"},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-3815\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"a\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"href\\":\\"http://localhost/rhel/vulnerability/cves/CVE-2019-3815\\",\\"children\\":\\"2\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-3815\\",\\"ref\\":null,\\"props\\":{\\"children\\":\\"Medium\\"},\\"_owner\\":null,\\"_store\\":{}}},{\\"title\\":{\\"type\\":\\"span\\",\\"key\\":\\"CVE-2019-3815\\",\\"ref\\":null,\\"props\\":{\\"children\\":[\\"\\",\\" \\",\\"On-hold\\"]},\\"_owner\\":null,\\"_store\\":{}}}],\\"isOpen\\":false,\\"selected\\":false},{\\"cells\\":[{\\"title\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":[{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Description\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"component\\":\\"p\\",\\"children\\":\\"A memory leak was discovered\\"},\\"_owner\\":null,\\"_store\\":{}}]},\\"_owner\\":null,\\"_store\\":{}}}],\\"parent\\":4}],\\"meta\\":{\\"page\\":1,\\"page_size\\":20,\\"total_items\\":248,\\"pages\\":10,\\"sort\\":\\"-public_date\\",\\"filter\\":\\"\\",\\"show_all\\":\\"false\\",\\"cvss_from\\":\\"\\",\\"cvss_to\\":\\"\\",\\"public_from\\":\\"\\",\\"public_to\\":\\"\\"},\\"isLoading\\":false}"`;
